File: Data Structures & Algorithms/maximum-subarray/submission-0.java
class Solution {
    public int maxSubArray(int[] nums) {
        int curSum = 0;
        int maxSum = nums[0];

        for (int n : nums) {
            curSum = Math.max(curSum, 0);
            curSum += n;
            maxSum = Math.max(curSum, maxSum);
        } 

        return maxSum;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-0.kt
class Solution {
    fun findMaxConsecutiveOnes(nums: IntArray): Int {
        var maxCount = 0
        var count = 0

        for (num in nums) {
            if (num == 1) {
                count++
                maxCount = Math.max(count, maxCount)
            } else {
                count = 0
            }
        }

        return maxCount
    }
}
